src/qibo/models/circuit.py

```diff
@@ -1012,6 +1022,27 @@ def __call__(self, initial_state=None, nshots=None):
         """Equivalent to ``circuit.execute``."""
         return self.execute(initial_state=initial_state, nshots=nshots)
 
+    @property
+    def raw(self) -> dict:
+        """Serialize to dictionary.
+
+        This is a thin wrapper over :meth:`Gate.raw`.
+        """
+        return {"queue": [gate.raw for gate in self.queue], "nqubits": self.nqubits}
```

Comment:
As a note, if I understood correctly what @ingoroth explained we should be saving not only the `circuit.queue` but also the `circuit.queue.moments` as the latter should be a more precise representation that could differ from the rebuild circuit by just adding the same gates back.

Reply:
According to @stavros11 the moments are built from what you have done with `add`, so there should not be any information loss.

Could you (or @ingoroth) provide an example of a circuit in which the `circuit.queue.moments` are not determined only by the addition sequence?

Reply:
Indeed, I think a given queue is mapped to a specific moment structure, because we don't provide options on how the moments are generated and by default each gate will be applied to the earliest possible moment. Therefore, if you save the queue and then you reconstruct the circuit you should end up with the same moments.

Reply:
If for now we don't have any options to play with the gate placement then they should be the same. One example would be instead of playing 2 gates in parallel in 2 qubits to play one gate and hte when it finishes play the other, I think. But, as we don't even are able to generate that circuit it should no be a problem for the serialiaze but I guess some people may want this flexibility when building circuits.

Reply:
Ok, then my current proposal would be to roll out serialization as is (just give me the time of writing tests).

The moment we will add the feature proposed (to support moments customization) we will upgrade serialization to support it as well.